app/pages/household/[houseId]/index.tsx

```diff
@@ -43,6 +56,36 @@ const Login = () => {
     console.log("Document written with ID: ", docRef.id);
   };
 
+  const handleDeleteHouse = async (event: any) => {
+    event.preventDefault();
+
+    const docRef = await deleteDoc(doc(db, "household", householdRef.id));
+
+    if (modalCheckboxRef.current !== null) {
+      modalCheckboxRef.current.checked = !modalCheckboxRef.current.checked;
+    }
+    console.log("House deleted");
+  };
+
+  const handleLeaveHouse = async (event: any) => {
+    event.preventDefault();
+    const currentUserId = currentUser?.userId;
+    const docRef = doc(db, "household", householdRef.id);
+    // max users 20 assumed
+    for (let i = 0; i< 20; i++){
```

Comment:
You might be able to use [FieldValue](https://firebase.google.com/docs/reference/android/com/google/firebase/firestore/FieldValue) arrayRemove to delete the user without checking if exists. See this https://stackoverflow.com/a/59745086/13177000